pyproject.toml

```diff
@@ -20,19 +21,18 @@ classifiers = [
   'Topic :: Scientific/Engineering'
 ]
 dependencies = [
-  'aiida-core[atomic_tools]~=2.5',
+  'aiida-core[atomic_tools]~=2.6',
   'aiida-shell>=0.5.3',
-  'dill',
   'numpy',
   'pint',
-  'pyscf[geomopt]~=2.2,<2.4'
+  'pyscf[geomopt] ~= 2.7'
 ]
 dynamic = ['description', 'version']
 keywords = ['aiida', 'workflows', 'pyscf']
 license = {file = 'LICENSE.txt'}
 name = 'aiida-pyscf'
 readme = 'README.md'
-requires-python = '>=3.9'
+requires-python = '>=3.9,<3.12'
```

Comment:
Thanks for looking after this plugin @ConradJohnston . Changes look fine and great that `pickle` can now be used. Was just wondering why you introduced this upper limit? There is a [long open discussion](https://discuss.python.org/t/requires-python-upper-limits/12663) on this practice on PyPA but the tendency seems to be advise against this practice. It is a complete topic, but in short this move is not as effective as you may think. If you were to release this change with a new version, dependency solvers running on Python 3.12 will simply revert to the latest release before the upper limit was imposed. If those versions are really incompatible with Python 3.12 and newer, the code would of course still fail. The only real reason would be to yank all previous versions from pip to prevent them from getting matched, which is of course not really a solution. Anyway, I would simply remove this upper limit.

Reply:
Thanks for this @sphuber - you make a good point. This is indeed counter to what I expected pip to do. 
I'll remove this cap.